src/sentry/static/sentry/app/views/discover/index.tsx

```diff
@@ -204,11 +201,16 @@ class DiscoverContainer extends React.Component<Props, State> {
     });
   };
 
-  renderNoAccess() {
+  renderNoAccess = () => {
+    const {router, organization} = this.props;
+
     return (
-      <AlertAsAny type="warning">{t("You don't have access to this feature")}</AlertAsAny>
+      <Redirect
```

Comment:
@Zylphrex We need to redirect depending on the org's feature availability of discover. 

- If the org has `organizations:discover-query`, we redirect them to `/organizations/:org_id/discover/queries/`.
- Otherwise, if the org has `organizations:discover-basic`, we redirect them to `/organizations/:org_id/discover/results/`.

I think you can make use of this pattern to accomplish it: https://github.com/getsentry/sentry/blob/2d94296414a0ffb5fd1634766595199695985f3b/src/sentry/static/sentry/app/views/eventsV2/index.tsx#L32-L39

Reply:
This redirect happens in the `renderDisabled` prop for the `Feature` below. Is this what you mean or did you mean something else?

Reply:
@Zylphrex The Feature component referred to below is referencing Discover v1 feature (`organizations:discover`).

The features I mentioned are for Discover v2, and we'd need to make the appropriate redirect. 

For example, on the team plan, `/organizations/:org_id/discover/queries/` is an empty "You don't have access to this feature" page.